moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/constraints_library.h

```diff
@@ -190,12 +190,13 @@ class ConstraintsLibrary
   const ConstraintApproximationPtr& getConstraintApproximation(const moveit_msgs::Constraints& msg) const;
 
 private:
-  ompl::base::StateStoragePtr constructConstraintApproximation(
-      const ModelBasedPlanningContextPtr& pcontext, const moveit_msgs::Constraints& constr_sampling,
-      const moveit_msgs::Constraints& constr_hard, const ConstraintApproximationConstructionOptions& options,
-      ConstraintApproximationConstructionResults& result);
+  ompl::base::StateStoragePtr
+  constructConstraintApproximation(ModelBasedPlanningContext* pcontext, const moveit_msgs::Constraints& constr_sampling,
+                                   const moveit_msgs::Constraints& constr_hard,
+                                   const ConstraintApproximationConstructionOptions& options,
+                                   ConstraintApproximationConstructionResults& result);
 
-  const PlanningContextManager& context_manager_;
+  ModelBasedPlanningContext* context_;
```

Comment:
I believe because we are now constructing the constraints library from inside the ModelBasedPlanningContext, with a call to this (https://github.com/ros-planning/moveit/pull/1428/commits/a9e91abf803b3102d038e68ef731512101092a1a#diff-d949b241fc2b4781b617fdb53844ece5R85).  Since `this` is naturally a raw pointer, I just kept it like that, and am a little hesitant to try to convert `this` to a reference or a shared pointer.